libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFSDKDatasharingHelper.m

```diff
@@ -68,6 +67,9 @@ - (void)setAppGroupEnabled:(BOOL)appGroupEnabled {
     NSUserDefaults *sharedDefaults = [[NSUserDefaults alloc] initWithSuiteName:self.appGroupName];
     [sharedDefaults setBool:appGroupEnabled forKey:kAppGroupEnabled];
     [sharedDefaults synchronize];
+    if(appGroupEnabled) {
```

Comment:
As part of this, can we amend the logic of this method to cover the adverse case?  I.e. if `appGroupEnabled` is already configured in the suite, *and* this is being set to `NO` here, throw a warning log that says e.g. "CAUTION YOU'RE ALMOST CERTAINLY GOING TO LOSE PERSISTED STATE IN YOUR APP BY REVERTING TO A NON-SHARED CONFIGURATION."

I'd actually consider adding an assertion failure in that case, since a) taking the action is fully in the developer's control, and b) going in that direction is almost certainly going to bork their application state.